tests/Signature/SignatureV4Test.php

```diff
@@ -277,6 +277,32 @@ public function testProvider()
     /**
      * @dataProvider testProvider
      */
+    public function testSignRequestUnsignedPayload($req, $sreq, $creq)
```

Comment:
Will this also include adding a new testprovider with 'UNSIGNED-PAYLOAD'  for testSignRequestUnsignedPayload() test and then just do a blackbox test?